src/FinfoMimeTypeDetector.php

```diff
@@ -53,9 +53,14 @@ public function __construct(
 
     public function detectMimeType(string $path, $contents): ?string
     {
-        $mimeType = is_string($contents)
-            ? (@$this->finfo->buffer($this->takeSample($contents)) ?: null)
-            : null;
+        $mimeType = null;
+        if (is_string($contents)) {
+            $mimeType = @$this->finfo->buffer($this->takeSample($contents));
+        } elseif (is_resource($contents) && get_resource_type($contents) === 'stream') {
+            $streamPosition = ftell($contents);
+            $mimeType = @$this->finfo->buffer(stream_get_contents($contents, $this->bufferSampleSize, 0));
```

Comment:
The bufferSampleSize can be null, but should have a default in this method call. I'm not sure what it should be. In the docs of node's file-type they show a value of 4100 which seems a bit much but the docs are written by an industry veteran, so might be ok as the default since it would be the entire file if not set.

Reply:
Adjusted - and now using a temporary stream for the detection since stream_copy_to_stream(), contrary to stream_get_contents()), should not reserve the whole buffer if unnecessary.
I'm currently torn between using `php://memory` and `php://temp` - the temp stream would have larger buffer support by eventually switching from memory to file handling.
However, it's cut-off is 2MB - which is larger than the now defined default.

Reply:
Switched to `php://temp` with custom memory cut-off limit set to the default buffer length. This way large custom buffers should not lead to memory hogging while the default will benefit from no file system operations.